aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -101,17 +102,17 @@ public S3FileIO(SerializableSupplier<S3Client> s3, AwsProperties awsProperties)
 
   @Override
   public InputFile newInputFile(String path) {
-    return S3InputFile.fromLocation(path, client(), awsProperties, metrics);
+    return S3InputFile.fromLocation(path, client(), awsProperties, metrics, bucketToAccessPointMapping);
   }
 
   @Override
   public OutputFile newOutputFile(String path) {
-    return S3OutputFile.fromLocation(path, client(), awsProperties, metrics, writeTags);
+    return S3OutputFile.fromLocation(path, client(), awsProperties, metrics, writeTags, bucketToAccessPointMapping);
```

Comment:
I am thinking about this a bit, it does not really scale if we keep adding new arguments to the constructor, we will basically have 1 new factory method per release. Traditionally we have been adding all the properties directly in `awsProperties`, and we did not really evaluate that option when adding `writeTags`. We should think about this a bit, I feel it is cleaner if we move all these new fields back to `AwsProperties`, it was designed to be the extension point for new configs without changing constructors in different places? Any thoughts?

Reply:
I agree with this, this should avoid changes to the constructor (or) introducing a new one. I will publish the PR for moving the new fields to `AwsProperties`. Thanks, for bringing this up.

Reply:
+1, Agree with this, changing constructors definitely doesn't seems to be scalable.

Reply:
I have published the PR: https://github.com/apache/iceberg/pull/4350